prow/git/v2/publisher.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+// Publisher knows how to publish local work to a remote
+type Publisher interface {
+	Commit(title, body string) error
+	ForcePush(branch string) error
```

Comment:
This assumes a single remote and that the local branch should track a remote branch with the same name. Are those assumptions OK?

Also I think we need a function to allow specifying the remote.